### PR TITLE
openblas, Python package cleanup

### PR DESCRIPTION
**Pull request by davidlange6**
Python updates: migration towards pip and removal of python not in cmssw-tool-conf (all unchanged for 4+ years)

otherwise, openblas included, which should replace the need/desire for atlas libraries. numpy updated. scipy moved to pip, and _should_ pick up the lin. algebra libraries using numpy's  configuration. I still need to confirm that. 

more checking needed before integrating. but lets see if there are issues at the cmssw level.

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: py2-pysqlite.spec [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-python setup.py build
-%install
-python setup.py install --prefix=%i
+### RPM external py2-pysqlite 2.8.3
+## INITENV +PATH PYTHONPATH %{i}/$PYTHON_LIB_SITE_PACKAGES
+
+
+%define pip_name pysqlite
+Requires: sqlite py2-setuptools 
+%define PipPreBuild tar xfz pysqlite-%{realversion}.tar.gz;rm pysqlite-%{realversion}.tar.gz;echo "include_dirs=$SQLITE_ROOT/include">>pysqlite-%{realversion}/setup.cfg;tar cfz pysqlite-%{realversion}.tar.gz pysqlite-%{realversion}
```
Comment (davidlange6) [inline py2-pysqlite.spec:7]: this is not completely safe btw. seems that the tarball gets left around when doing multiple attempts at building.

**Comment by smuzaffar** [comment]
actual error is 
```
Trying to install the rpm package external+py2-nose+1.3.7-cms just built.
Checking local path dependency for rpm package external+py2-nose+1.3.7-cms just build.
Error: Local path dependency found for external+py2-nose+1.3.7-cms
  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/slc6_amd64_gcc530/external/python/2.7.11-oenich2/bin/python
```

**Comment by davidlange6** [comment]
weird - I should have had the same problem.. ok, that is easy to fix assuming I can reproduce it.

**Comment by davidlange6** [comment]
please test

(my mistake it seems)

**Comment by davidlange6** [comment]
with this fix I can create a python env easily.. (ok, it leaves out some of the python packages not installed using pip)

[dlange@cmsdev01 170323t2]$ source slc6_amd64_gcc530/external/py2-pippkgs_depscipy/3.0-cms8/etc/profile.d/init.csh 
[dlange@cmsdev01 170323t2]$ which python
/build/dlange/170323t2/slc6_amd64_gcc530/external/python/2.7.11-cms/bin/python
[dlange@cmsdev01 170323t2]$ python
Python 2.7.11 (default, Mar 23 2017, 18:11:22) 
[GCC 5.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> import matplotlib
>>> import keras
Using Theano backend.
>>>

**Comment by smuzaffar** [comment]
please test

**Comment by davidlange6** [comment]
please test

(the test results are gone?)
i checked numpy in my area, it looks like it is set up correctly

**Comment by davidlange6** [comment]
please test



On 24 Mar 2017, at 18:08, cmsbuild <notifications@github.com<mailto:notifications@github.com>> wrote:


-1

Tested at: 0e98a78<https://github.com/cms-sw/cmsdist/commit/0e98a780b534ab581910e6cdea1f465ea935c0cf>

  *   Build:

I found an error when building:

+ sed -ideleteme 's|#!.*/bin/python|#!/usr/bin/env python|' /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/BUILDROOT/41cfad3ae9fb2eb48557307da2eac2d3/opt/cmssw/slc6_amd64_gcc530/external/py2-numpy/1.12.1/bin/f2py '/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/BUILDROOT/41cfad3ae9fb2eb48557307da2eac2d3/opt/cmssw/slc6_amd64_gcc530/external/py2-numpy/1.12.1/lib/python*/site-packages/numpy-*/EGG-INFO/scripts/f2py' '/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/BUILDROOT/41cfad3ae9fb2eb48557307da2eac2d3/opt/cmssw/slc6_amd64_gcc530/external/py2-numpy/1.12.1/lib/python*/site-packages/numpy-*/numpy/core/tests/test_arrayprint.py' '/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/BUILDROOT/41cfad3ae9fb2eb48557307da2eac2d3/opt/cmssw/slc6_amd64_gcc530/external/py2-numpy/1.12.1/lib/python*/site-packages/numpy-*/numpy/distutils/from_template.py' '/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/BUILDROOT/41cfad3ae9fb2eb48557307da2eac2d3/opt/cmssw/slc6_amd64_gcc530/external/py2-numpy/1.12.1/lib/python*/site-packages/numpy-*/numpy/distutils/conv_template.py'
 sed: can't read /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/BUILDROOT/41cfad3ae9fb2eb48557307da2eac2d3/opt/cmssw/slc6_amd64_gcc530/external/py2-numpy/1.12.1/lib/python*/site-packages/numpy-*/EGG-INFO/scripts/f2py: No such file or directory
 sed: can't read /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/BUILDROOT/41cfad3ae9fb2eb48557307da2eac2d3/opt/cmssw/slc6_amd64_gcc530/external/py2-numpy/1.12.1/lib/python*/site-packages/numpy-*/numpy/core/tests/test_arrayprint.py: No such file or directory
 sed: can't read /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/BUILDROOT/41cfad3ae9fb2eb48557307da2eac2d3/opt/cmssw/slc6_amd64_gcc530/external/py2-numpy/1.12.1/lib/python*/site-packages/numpy-*/numpy/distutils/from_template.py: No such file or directory
 sed: can't read /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/BUILDROOT/41cfad3ae9fb2eb48557307da2eac2d3/opt/cmssw/slc6_amd64_gcc530/external/py2-numpy/1.12.1/lib/python*/site-packages/numpy-*/numpy/distutils/conv_template.py: No such file or directory
 error: Bad exit status from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/rpm-tmp.UjvdJX (%install)


 RPM build errors:
 Bad exit status from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp/rpm-tmp.UjvdJX (%install)



You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2915/18676/summary.html

**Comment by davidlange6** [comment]
strange, will try again to reproduce locally...(but means I won't merge tonight as planned.)

**Comment by smuzaffar** [comment]
I restarted the tests in jenkins and it looks good now ... externals are build and now cmssw is being tested.

**Comment by smuzaffar** [comment]
looks good now, only one unit test failed and all relval tests passed.
I am merging it now for next IB (which will start in 30 mins)

**Comment by davidlange6** [comment]
indeed, my local test did finish ok. Thanks!

**Pull request merged**